Device ID: 183

Received 4 sensors: [{'sensor_id': 31, 'area_id': 44, 'type': 'NPK', 'name': 'NPK Sensor', 'unit': 'mg/kg', 'threshold': {'K': {'max': 400, 'min': 100}, 'N': {'max': 300, 'min': 34}, 'P': {'max': 300, 'min': 75}}, 'domain': {'max': 1000, 'min': 0}, 'greenhouse_id': 41}, {'sensor_id': 16, 'area_id': 26, 'type': 'pH', 'name': 'pH Sensor', 'unit': 'pH', 'threshold': {'max': 6.5, 'min': 5}, 'domain': {'max': 10.0, 'min': 3.0}, 'greenhouse_id': 31}, {'sensor_id': 22, 'area_id': 33, 'type': 'pH', 'name': 'pH Sensor', 'unit': 'pH', 'threshold': {'max': 9.5, 'min': 6}, 'domain': {'max': 10.0, 'min': 3.0}, 'greenhouse_id': 35}, {'sensor_id': 30, 'area_id': 41, 'type': 'SoilMoisture', 'name': 'Soil Moisture Sensor', 'unit': '%', 'threshold': {'max': 80, 'min': 70}, 'domain': {'max': 100, 'min': 0}, 'greenhouse_id': 39}]

Subscribed to greenhouse_41/area_44/sensor_31
Connected to mosquitto with result code 0
Subscribed to greenhouse_41/area_44/event/sensor_31
Subscribed to greenhouse_31/area_26/sensor_16
Subscribed to greenhouse_31/area_26/event/sensor_16
Subscribed to greenhouse_35/area_33/sensor_22
Subscribed to greenhouse_35/area_33/event/sensor_22
Subscribed to greenhouse_39/area_41/sensor_30
Subscribed to greenhouse_39/area_41/event/sensor_30
Started periodic sensor checking thread

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 140.73, 'P': 205.58, 'K': 194.15}, 'u': 'mg/kg', 't': 60}}
WARNING: Failed to get when the next value of NPK (sensor_31) is expected to arrive
Missing key in the message: 'next_value'
Periodic sensor check started
Periodic sensor check completed

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 140.33, 'P': 204.4, 'K': 193.88}, 'u': 'mg/kg', 't': 120}}
The next value of NPK (sensor_31) is expected to be received at: 180.0
The next value of NPK (sensor_31) is expected to be: {'N': 139.93000000000004, 'P': 203.22, 'K': 193.60999999999999} mg/kg
Wait to see another value to start the decision making for sensor_31 (NPK)
Periodic sensor check started
Periodic sensor check completed

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 139.09, 'P': 205.48, 'K': 192.23}, 'u': 'mg/kg', 't': 180}}
The next value of NPK (sensor_31) is expected to be received at: 240.0
The next value of NPK (sensor_31) is expected to be: {'N': 138.41, 'P': 205.0533333333333, 'K': 191.49999999999997} mg/kg
The measured value (139.09 mg/kg) and the next prediction (138.41 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c585d0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (205.48 mg/kg) and the next prediction (205.0533333333333 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c5a250>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (192.23 mg/kg) and the next prediction (191.49999999999997 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c58690>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 138.97, 'P': 205.45, 'K': 192.62}, 'u': 'mg/kg', 't': 240}}
The next value of NPK (sensor_31) is expected to be received at: 300.0
The next value of NPK (sensor_31) is expected to be: {'N': 138.15, 'P': 205.39999999999998, 'K': 191.66} mg/kg
The measured value (138.97 mg/kg) and the next prediction (138.15 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c5b590>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (205.45 mg/kg) and the next prediction (205.39999999999998 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c4a410>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (192.62 mg/kg) and the next prediction (191.66 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c49d90>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 137.94, 'P': 204.66, 'K': 191.29}, 'u': 'mg/kg', 't': 300}}
The next value of NPK (sensor_31) is expected to be received at: 360.0
The next value of NPK (sensor_31) is expected to be: {'N': 137.32999999999998, 'P': 204.87700000000004, 'K': 190.73999999999998} mg/kg
The measured value (137.94 mg/kg) and the next prediction (137.32999999999998 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a5e710>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (204.66 mg/kg) and the next prediction (204.87700000000004 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a649d0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (191.29 mg/kg) and the next prediction (190.73999999999998 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a65190>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 138.48, 'P': 203.54, 'K': 192.07}, 'u': 'mg/kg', 't': 360}}
The next value of NPK (sensor_31) is expected to be received at: 420.0
The next value of NPK (sensor_31) is expected to be: {'N': 137.40266666666668, 'P': 203.9066666666667, 'K': 190.92866666666666} mg/kg
The measured value (138.48 mg/kg) and the next prediction (137.40266666666668 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c51650>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (203.54 mg/kg) and the next prediction (203.9066666666667 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c5b3d0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (192.07 mg/kg) and the next prediction (190.92866666666666 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a5dad0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 136.49, 'P': 203.64, 'K': 190.39}, 'u': 'mg/kg', 't': 420}}
The next value of NPK (sensor_31) is expected to be received at: 480.0
The next value of NPK (sensor_31) is expected to be: {'N': 136.35142857142856, 'P': 203.4842857142857, 'K': 190.11285714285714} mg/kg
The measured value (136.49 mg/kg) and the next prediction (136.35142857142856 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c6d010>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (203.64 mg/kg) and the next prediction (203.4842857142857 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a67a50>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (190.39 mg/kg) and the next prediction (190.11285714285714 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a5c510>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 137.08, 'P': 203.21, 'K': 189.72}, 'u': 'mg/kg', 't': 480}}
The next value of NPK (sensor_31) is expected to be received at: 540.0
The next value of NPK (sensor_31) is expected to be: {'N': 136.0882142857143, 'P': 203.04857142857142, 'K': 189.35071428571428} mg/kg
The measured value (137.08 mg/kg) and the next prediction (136.0882142857143 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c59710>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (203.21 mg/kg) and the next prediction (203.04857142857142 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a6cc90>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (189.72 mg/kg) and the next prediction (189.35071428571428 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c52d90>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 137.29, 'P': 201.91, 'K': 189.44}, 'u': 'mg/kg', 't': 540}}
The next value of NPK (sensor_31) is expected to be received at: 600.0
The next value of NPK (sensor_31) is expected to be: {'N': 136.05555555555557, 'P': 202.22111111111113, 'K': 188.79194444444443} mg/kg
The measured value (137.29 mg/kg) and the next prediction (136.05555555555557 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c4a8d0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (201.91 mg/kg) and the next prediction (202.22111111111113 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c6f850>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (189.44 mg/kg) and the next prediction (188.79194444444443 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c522d0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 139.13, 'P': 203.29, 'K': 187.95}, 'u': 'mg/kg', 't': 600}}
The next value of NPK (sensor_31) is expected to be received at: 660.0
The next value of NPK (sensor_31) is expected to be: {'N': 136.7986666666667, 'P': 202.25133333333332, 'K': 187.86266666666666} mg/kg
The measured value (139.13 mg/kg) and the next prediction (136.7986666666667 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c50950>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (203.29 mg/kg) and the next prediction (202.25133333333332 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c6fc90>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (187.95 mg/kg) and the next prediction (187.86266666666666 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a66690>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 139.22, 'P': 202.73, 'K': 186.7}, 'u': 'mg/kg', 't': 660}}
The next value of NPK (sensor_31) is expected to be received at: 720.0
The next value of NPK (sensor_31) is expected to be: {'N': 137.64600000000002, 'P': 202.08733333333333, 'K': 186.73333333333332} mg/kg
The measured value (139.22 mg/kg) and the next prediction (137.64600000000002 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c4b9d0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (202.73 mg/kg) and the next prediction (202.08733333333333 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a75b50>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (186.7 mg/kg) and the next prediction (186.73333333333332 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a5c510>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 138.19, 'P': 203.54, 'K': 187.7}, 'u': 'mg/kg', 't': 720}}
The next value of NPK (sensor_31) is expected to be received at: 780.0
The next value of NPK (sensor_31) is expected to be: {'N': 138.07533333333336, 'P': 202.12266666666665, 'K': 186.4286666666667} mg/kg
The measured value (138.19 mg/kg) and the next prediction (138.07533333333336 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c49390>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (203.54 mg/kg) and the next prediction (202.12266666666665 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a75b50>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (187.7 mg/kg) and the next prediction (186.4286666666667 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c51950>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 138.79, 'P': 202.89, 'K': 188.73}, 'u': 'mg/kg', 't': 780}}
The next value of NPK (sensor_31) is expected to be received at: 840.0
The next value of NPK (sensor_31) is expected to be: {'N': 138.55666666666664, 'P': 202.24333333333328, 'K': 186.50799999999998} mg/kg
The measured value (138.79 mg/kg) and the next prediction (138.55666666666664 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a76590>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (202.89 mg/kg) and the next prediction (202.24333333333328 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a5cd50>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (188.73 mg/kg) and the next prediction (186.50799999999998 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c51a10>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 138.43, 'P': 202.91, 'K': 187.41}, 'u': 'mg/kg', 't': 840}}
The next value of NPK (sensor_31) is expected to be received at: 900.0
The next value of NPK (sensor_31) is expected to be: {'N': 138.88199999999998, 'P': 202.53666666666666, 'K': 186.3966666666667} mg/kg
The measured value (138.43 mg/kg) and the next prediction (138.88199999999998 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a75490>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (202.91 mg/kg) and the next prediction (202.53666666666666 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a83290>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (187.41 mg/kg) and the next prediction (186.3966666666667 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c6e7d0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 139.79, 'P': 202.25, 'K': 188.72}, 'u': 'mg/kg', 't': 900}}
The next value of NPK (sensor_31) is expected to be received at: 960.0
The next value of NPK (sensor_31) is expected to be: {'N': 139.5126666666667, 'P': 202.5246666666667, 'K': 186.802} mg/kg
The measured value (139.79 mg/kg) and the next prediction (139.5126666666667 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c5a0d0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (202.25 mg/kg) and the next prediction (202.5246666666667 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c51950>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (188.72 mg/kg) and the next prediction (186.802 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a66010>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 137.79, 'P': 201.92, 'K': 189.53}, 'u': 'mg/kg', 't': 960}}
The next value of NPK (sensor_31) is expected to be received at: 1020.0
The next value of NPK (sensor_31) is expected to be: {'N': 139.36399999999995, 'P': 202.24266666666668, 'K': 187.91066666666666} mg/kg
The measured value (137.79 mg/kg) and the next prediction (139.36399999999995 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a95750>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (201.92 mg/kg) and the next prediction (202.24266666666668 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32a66590>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (189.53 mg/kg) and the next prediction (187.91066666666666 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c6c610>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 135.71, 'P': 201.03, 'K': 188.52}, 'u': 'mg/kg', 't': 1020}}
The next value of NPK (sensor_31) is expected to be received at: 1080.0
The next value of NPK (sensor_31) is expected to be: {'N': 137.89866666666666, 'P': 201.73933333333332, 'K': 188.33666666666664} mg/kg
The measured value (135.71 mg/kg) and the next prediction (137.89866666666666 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c590d0>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (201.03 mg/kg) and the next prediction (201.73933333333332 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Failed to send Telegram message: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot7794925862:AAEeWdf4lWJeLpexbiMXqeBVVDRXYDEzIu4/sendMessage (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f8c32c51b90>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
The measured value (188.52 mg/kg) and the next prediction (188.33666666666664 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Telegram message sent successfully to user 788476502

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 134.54, 'P': 200.98, 'K': 189.79}, 'u': 'mg/kg', 't': 1081}}
The next value of NPK (sensor_31) is expected to be received at: 1141.111111111111
The next value of NPK (sensor_31) is expected to be: {'N': 136.16881665904305, 'P': 201.2724878618928, 'K': 189.2191108068392} mg/kg
The measured value (134.54 mg/kg) and the next prediction (136.16881665904305 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Telegram message sent successfully to user 788476502
The measured value (200.98 mg/kg) and the next prediction (201.2724878618928 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Telegram message sent successfully to user 788476502
The measured value (189.79 mg/kg) and the next prediction (189.2191108068392 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Telegram message sent successfully to user 788476502

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 136.29, 'P': 201.28, 'K': 187.94}, 'u': 'mg/kg', 't': 1141}}
The next value of NPK (sensor_31) is expected to be received at: 1201.111111111111
The next value of NPK (sensor_31) is expected to be: {'N': 135.37187542511072, 'P': 200.73151052126627, 'K': 189.27812558098466} mg/kg
The measured value (136.29 mg/kg) and the next prediction (135.37187542511072 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Periodic sensor check started
Periodic sensor check completed
Telegram message sent successfully to user 788476502
The measured value (201.28 mg/kg) and the next prediction (200.73151052126627 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Telegram message sent successfully to user 788476502
The measured value (187.94 mg/kg) and the next prediction (189.27812558098466 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Telegram message sent successfully to user 788476502

Received: {'bn': 'greenhouse_41/area_44/sensor_31', 'e': {'n': 'NPK', 'v': {'N': 137.92, 'P': 201.67, 'K': 186.23}, 'u': 'mg/kg', 't': 1201}}
Periodic sensor check started
The next value of NPK (sensor_31) is expected to be received at: 1261.111111111111
Periodic sensor check completed
The next value of NPK (sensor_31) is expected to be: {'N': 135.78530627704066, 'P': 200.75706914936003, 'K': 188.3557899977533} mg/kg
The measured value (137.92 mg/kg) and the next prediction (135.78530627704066 mg/kg) of NPK - N (sensor_31) are inside the range [34, 300]
Telegram message sent successfully to user 788476502
The measured value (201.67 mg/kg) and the next prediction (200.75706914936003 mg/kg) of NPK - P (sensor_31) are inside the range [75, 300]
Telegram message sent successfully to user 788476502
The measured value (186.23 mg/kg) and the next prediction (188.3557899977533 mg/kg) of NPK - K (sensor_31) are inside the range [100, 400]
Telegram message sent successfully to user 788476502
